sdk-tests/src/test/java/io/dapr/it/testcontainers/workflows/crossapp/WorkflowsCrossAppCallActivityIT.java

```diff
@@ -0,0 +1,327 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dapr.it.testcontainers.workflows.crossapp;
+
+import io.dapr.testcontainers.Component;
+import io.dapr.testcontainers.DaprContainer;
+import io.dapr.testcontainers.DaprLogLevel;
+import io.dapr.workflows.client.DaprWorkflowClient;
+import io.dapr.workflows.client.WorkflowInstanceStatus;
+import io.dapr.workflows.client.WorkflowRuntimeStatus;
+import io.dapr.config.Properties;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.MountableFile;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Map;
+
+import static io.dapr.it.testcontainers.ContainerConstants.DAPR_RUNTIME_IMAGE_TAG;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import org.testcontainers.images.builder.Transferable;
+
+/**
+ * Cross-App Pattern integration test.
+ *
+ * This test demonstrates the cross-app pattern by:
+ * 1. Starting 3 Dapr containers (crossapp-worker, app2, app3)
+ * 2. Launching Java processes that register workflows/activities in separate apps
+ * 3. Executing a cross-app workflow
+ * 4. Asserting successful completion
+ */
+@Testcontainers
+@Tag("testcontainers")
+public class WorkflowsCrossAppCallActivityIT {
+
+  private static final Network DAPR_NETWORK = Network.newNetwork();
+
+  // Main workflow orchestrator container
+  @Container
+  private static final DaprContainer MAIN_WORKFLOW_CONTAINER = new DaprContainer(DAPR_RUNTIME_IMAGE_TAG)
+      .withAppName("crossapp-worker")
+      .withNetwork(DAPR_NETWORK)
+      .withNetworkAliases("main-workflow-sidecar")
+      .withComponent(new Component("kvstore", "state.in-memory", "v1",
+          Map.of("actorStateStore", "true")))
+      .withComponent(new Component("pubsub", "pubsub.in-memory", "v1", Collections.emptyMap()))
+      .withDaprLogLevel(DaprLogLevel.DEBUG)
+      .withLogConsumer(outputFrame -> System.out.println("MAIN_WORKFLOW: " + outputFrame.getUtf8String()))
+      .withAppChannelAddress("host.testcontainers.internal");
+
+  // App2 container for App2TransformActivity - using GenericContainer for custom ports
+  @Container
+  private static final GenericContainer<?> APP2_CONTAINER = new GenericContainer<>(DAPR_RUNTIME_IMAGE_TAG)
+      .withNetwork(DAPR_NETWORK)
+      .withNetworkAliases("app2-sidecar")
+      .withExposedPorts(3501, 50002)
+      .withCommand("./daprd", 
+                   "--app-id", "app2",
+                   "--dapr-listen-addresses=0.0.0.0",
```

Comment:
Im tweaking this bit, so dont review on these specific lines. will update the `DaprContainer.java` to clean this container definition up.